### ISSUE TITLE
IIS Express asp.net web app 403.14 error

### ISSUE THREAD

**Issue by vilmantas** [warrenbuckley/IIS-Express-Code#29]
I tried running my project with this extension but i keep getting 403.14 errors even though when it is built/ran with visual studio 2017 iis express it runs smoothly. I checked configs from both results and they are the same (apart from generated site name and config locations - one being .vs folder and other iis express default folder).  Is there any way to see what Visual Studio does different when launching IIS Express compared to this extension?

Also index file is reachable when browsing with localhost:port/wwwroot/index.html

**Comment by DonInMedford** [comment]
I have the same problem. It happened a few days ago.  They also updated IIS Express the same time, not sure if that was related or not. It worked excellent before that. I was able to sort of fix it by going into the IIS Express folder in My Documents and editing the applicationhost.config line 217 directorybrowse enabled = true. It's not a complete fix as I have to manually open from the folder to show the website, but it now works instead of the error.

**Comment by DonInMedford** [comment]
Hello Warren, your extension did work very well serving up html files for a few weeks. Then stopped working giving the error listed above. I noticed that the IIS Express was updated about the same time. On the error page for 403.14 it tells me to look at the config file and change directorybrowse enabled = true. So that is what I did. Then the directory would show and I could launch the index.html pages by clicking into the directory. I am on a Windows 10 Pro 64 bit machine. I installed VS Code using the regular download on code.visualstudio.com about three weeks ago. I am new at all this so it's most likely an error on my part. If no one else has an issue I would not put anymore effort into it. VSCode and IIS Express was just an experiment, and I have gone back to using sublime and webpacks dev server. I do appreciate all your work, if you do need to have me reinstall the VS code, I would be willing to do so and give you any info. I still have IIS Express installed and it works. Thank you again for your work. Don

**Comment by jeremyhoesly** [comment]
I am also encountering the same behavior in my ASP.NET Core MVC projects. I confirmed the error can be reproduced with a new ASP.NET Core MVC project. It opens fine using Kestrel but when using the IIS Express extension, a 403.14 error occurs. I noticed that the extension is not starting dotnet.exe beforehand like Visual Studio 2017. I am running Visual Studio Code 1.14.2 and Visual Studio Professional 2017 15.2 (26430.16).

**Comment by jason2li** [comment]
Any updates on this?

**Comment by warrenbuckley** [comment]
@jason2li & @jeremyhoesly apologies for the lack of updates from me. I have not used any .NET Core WebApps yet & thus never run into this problem.

I will try and see if I can reproduce this today

**Comment by kevingates** [comment]
Any updates on this?

**Comment by ORoberts221** [comment]
I have just encountered this issue - it was caused by accidentally trying to run the wrong directory. 

I have a solution directory with several project directories within it, one of which is the website I want to run. I was trying to open and run the solution directory, which caused IIS to think I was trying to run a folder which it isn't configured to do by default (so it results in this 403 error). By adding the same config setting @DonInMedford used I was able to get IIS to open the solution directory - where I could click in to a project and view individual .html files. 

This is when I realised it was opening the (incorrect) parent directory. Opening the project directory (the one that contains the web.config) in VS Code and using IIS Start Website caused the project to run just fine.

**Comment by bellamira** [comment]
Wondering if this is still an issue or if maybe I am doing something wrong. I can run my app in Visual Studio 2019 using an IIS debug configuration, but I can't seem to get the same functionality with this extension in VS Code. I get the 403.14 error.

I copied the contents of the `applicationhost.config` from the project's .vs directory into the config that this extension seems to use (C:\Users\me\Documents\IISExpress\config\applicationhost.config) but it still does not work.

I also set `directorybrowse enabled=true` but that really does not help me because none of the folders I open run the site, either. 

I'm a little out of my element here, so if there is any information I can post from my config file for you to help me, I am more than willing and would greatly appreciate it! Thank you!
